drivers/mbox/mbox_ti_secproxy.c

```diff
@@ -359,28 +351,54 @@ static DEVICE_API(mbox, secproxy_mailbox_driver_api) = {
 	.set_enabled = secproxy_mailbox_set_enabled,
 };
 
-#define MAILBOX_INSTANCE_DEFINE(idx)                                                               \
-	static struct secproxy_mailbox_data secproxy_mailbox_##idx##_data;                         \
-	const static struct secproxy_mailbox_config secproxy_mailbox_##idx##_config = {            \
-		DEVICE_MMIO_NAMED_ROM_INIT_BY_NAME(target_data, DT_DRV_INST(idx)),                 \
-		DEVICE_MMIO_NAMED_ROM_INIT_BY_NAME(rt, DT_DRV_INST(idx)),                          \
-		DEVICE_MMIO_NAMED_ROM_INIT_BY_NAME(scfg, DT_DRV_INST(idx)),                        \
-		.irq = DT_INST_IRQN(idx),                                                          \
-	};                                                                                         \
-	static int secproxy_mailbox_##idx##_init(const struct device *dev)                         \
-	{                                                                                          \
-		DEVICE_MMIO_NAMED_MAP(dev, target_data, K_MEM_CACHE_NONE);                         \
-		DEVICE_MMIO_NAMED_MAP(dev, rt, K_MEM_CACHE_NONE);                                  \
-		DEVICE_MMIO_NAMED_MAP(dev, scfg, K_MEM_CACHE_NONE);                                \
-		IRQ_CONNECT(DT_INST_IRQN(idx), DT_INST_IRQ(idx, priority), secproxy_mailbox_isr,   \
-			    DEVICE_DT_INST_GET(idx),                                               \
-			    COND_CODE_1(DT_INST_IRQ_HAS_CELL(idx, flags),			\
-				(DT_INST_IRQ(idx, flags)), (0)));       \
-		return 0;                                                                          \
-	}                                                                                          \
-	DEVICE_DT_INST_DEFINE(idx, secproxy_mailbox_##idx##_init, NULL,                            \
-			      &secproxy_mailbox_##idx##_data, &secproxy_mailbox_##idx##_config,    \
-			      PRE_KERNEL_1, CONFIG_MBOX_TI_SECURE_PROXY_PRIORITY,                  \
+#define SECPROXY_THREAD_ISR(i, idx)                                                               \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(idx, DT_CAT(rx_, i)),                                    \
+		(                                                                                 \
+			static void secproxy_mailbox_isr_##idx##_##i(const struct device *dev)    \
+			{                                                                         \
+				secproxy_mailbox_isr(dev, i);                                     \
+			}                                                                         \
+		),                                                                                \
+		())
+
+/* Generate IRQ number or 0 for array initialization */
+#define SECPROXY_IRQ_OR_ZERO(i, inst)                                                             \
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(inst, DT_CAT(rx_, i)),                                   \
+		(DT_INST_IRQ_BY_NAME(inst, DT_CAT(rx_, i), irq)),                                 \
+		(0))
```

Comment:
is it possible to have another placeholder - like a negative value? since 0 is a valid IRQ.